### ISSUE TITLE
Buffered streams 'COMPLETED'

### ISSUE THREAD

**Issue by tetricky** [realOxy/M3UAndroid#69]
New to this project. It's a very nice way of handling playlists. Much superior to my experience in VLC. Thank you for it. 

When playing a stream, as part of a playlist, if the stream buffers it will time out, and show as completed. I understand that is useful and intended behavior.  There are occasions where it would be preferred to retry the stream ad infinitum. If listening to or watching a stream that regularly becomes interrupted, in it's current form (as per my understanding) I can find no way to automatically reconnect after a time out.

Is there an existing setting that makes this possible? If not could I put it forward as a suggestion?

**Comment by realOxy** [comment]
Hi @tetricky, have you checked "auto reconnect" preference in settings -> optional player features?

**Comment by tetricky** [comment]
Yes, I have done that...streams play the first segment, until a point where they have an extended period of waiting to buffer, and then show as completed and don't appear to retry.

**Comment by realOxy** [comment]
OK, I will try adjusting behavior of auto reconnect. it will always retry not only when playback gave error but also when playback complete.

**Comment by tetricky** [comment]
I do think your existing behaviour is 'correct'. Would it be possible to add some sort of toggle for 'always retry' to override that..so that people who want normal behaviour can continue to do so by leaving it unchecked?

**Comment by realOxy** [comment]
There will be 3 selection for "reconnect mode(renamed from auto reconnect)":
- never
- only when failed
- always

**Comment by tetricky** [comment]
1.14.0-alpha13 has just come through on Fdroid for me. When auto reconnect is set to 'ALWAYS' streams now continue restarting after what would previously be halted as 'COMPLETED'.

So that looks like the behaviour I was looking for. Thank you very much!
